youtube_dl/extractor/funimation.py

```diff
@@ -0,0 +1,97 @@
+    _USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/601.2.7 (KHTML, like Gecko) Version/9.0.1 Safari/601.2.7'
+
+    _TEST = {
+        'url': 'http://www.funimation.com/shows/air/videos/official/breeze',
+        'info_dict': {
+            'id': 'AIRENG0001',
+            'title': 'Air - Breeze',
+            'ext': 'mp4',
+        },
+        'params': {
+            'format': 'lower',
+        }
+    }
+
+    def _real_initialize(self):
+        self._login()
+
+    def _login(self):
+        (username, password) = self._get_login_info()
+        if username is None:
+            return
+        self.report_login()
+        login_url = 'http://www.funimation.com/login'
+        data = urlencode_postdata({
+            'formname': 'loginForm2',
+            'email_field': username,
+            'password_field': password,
+        })
+        login_request = sanitized_Request(login_url, data)
+        login_request.add_header('Content-Type', 'application/x-www-form-urlencoded')
+        try:
+            login = self._download_webpage(login_request, login_url)
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
+                raise ExtractorError('%s is not available in your region.' % self.IE_NAME, expected=True)
+        if re.search(r'<meta property="og:url" content="http://www.funimation.com/login"/>', login) is not None:
+                raise ExtractorError('Unable to login, wrong username or password', expected=True)
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        request = sanitized_Request(url)
+        request.add_header('User-Agent', self._USER_AGENT)
+        try:
+            webpage = self._download_webpage(request, url)
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
+                raise ExtractorError('%s is not available in your region.' % self.IE_NAME, expected=True)
+        if re.search(r'"sdUrl":"http', webpage) is None:
+                raise ExtractorError('You are not logged-in or the stream requires subscription.', expected=True)
+        self.report_extraction(url)
+
+        fallback_title = video_id
+        video_base = self._search_regex(r'"sdUrl":"(.+?)(?:-1500K.mp4)"', webpage, 'video_base')
+        video_auth = self._search_regex(r'"authToken":"(.+?)"', webpage, 'video_auth')
+        video_id = self._search_regex(r'"FUNImationID":"(.+?)"', webpage, 'video_id')
+        video_base = video_base.replace('\\', '')
+
+        # lower might be better as the default format, it is the most supported.
+        formats_map = [
+            ('lowest', 480, 1000),
+            ('lower', 480, 1500),
+            ('low', 480, 2000),
+            ('medium', 720, 2500),
+            ('high', 720, 3500),
+            ('highest', 1080, 4000),
```

Comment:
This can be easily detected from m3u8 that's served in `sdUrl` for default youtube-dl UA.